### PR TITLE
WebRequest/ServicePoint/WebClient Obsoletion Follow-Up

### LINKED ISSUE #41759: WebRequest/ServicePoint/WebClient Obsoletion Follow-Up
During a PR review of #41526, which would have ported #41400 into the `release/5.0` branch, there was some PR feedback.  Because that port PR was closed without merging, that feedback still needs to be addressed in `master`.

1. The message defined for the obsoletion needs to provide more context. See [this comment](https://github.com/dotnet/runtime/pull/41526#discussion_r479487823).
1. The obsoletion needs to be added to the [list of obsoletions](https://github.com/dotnet/runtime/blob/master/docs/project/list-of-obsoletions.md) document
1. We should add comments into the csproj files where the diagnostic is set to `NoWarn`. See [this comment](https://github.com/dotnet/runtime/pull/41526#discussion_r479489975), but also apply that to the other csproj files too

This obsoletion is [currently being absorbed](https://github.com/dotnet/aspnetcore/pull/25399#issuecomment-685992001) in the aspnetcore repo, and improving these details might have helped smooth that experience out.

### PR DESCRIPTION
**Pull request by jeffhandley**
Fixes #41759

During a PR review of #41526, which would have ported #41400 into the release/5.0 branch, there was some PR feedback. Because that port PR was closed without merging, that feedback went unaddressed.

- The message defined for the obsoletion needs to provide more context
- The obsoletion needs to be added to the document listing our `SYSLIB` obsoletions
- Add comments into the csproj files where the diagnostic is set to NoWarn

### CODE REVIEW

File: docs/project/list-of-diagnostics.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Currently the identifiers `SYSLIB0001` through `SYSLIB0999` are carved out for o
 |  __`SYSLIB0011`__ | `BinaryFormatter` serialization is obsolete and should not be used. See https://aka.ms/binaryformatter for recommended alternatives. |
 |  __`SYSLIB0012`__ | Assembly.CodeBase and Assembly.EscapedCodeBase are only included for .NET Framework compatibility. Use Assembly.Location instead. |
 |  __`SYSLIB0013`__ | Uri.EscapeUriString can corrupt the Uri string in some cases. Consider using Uri.EscapeDataString for query string components instead. |
+|  __`SYSLIB0014`__ | WebRequest, HttpWebRequest, ServicePoint, and WebClient are obsolete. Use HttpClient instead. |
```
Comment (stephentoub) [inline docs/project/list-of-diagnostics.md:29]: I was wondering previously why this was missing :)

File: src/libraries/System.Net.Requests/src/System/Net/FileWebRequest.cs [hunk 2]
```diff
@@ -25,7 +25,7 @@ public class FileWebRequest : WebRequest, ISerializable
         private bool _syncHint;
         private int _aborted;
 
-#pragma warning disable SYSLIB0014
+#pragma warning disable SYSLIB0014 // WebRequest, HttpWebRequest, ServicePoint, and WebClient are obsolete. Use HttpClient instead.
```
Comment (stephentoub) [inline src/libraries/System.Net.Requests/src/System/Net/FileWebRequest.cs:28]: Should we just suppress SYSLIB0014 in System.Net.Requests at the project level?  This assembly is entirely about providing these types.  Same for System.Net.ServicePoint.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Net.Requests/src/System/Net/FileWebRequest.cs (hunk 2 above)
Comment (jeffhandley) [reply src/libraries/System.Net.Requests/src/System/Net/FileWebRequest.cs:28]: I'm good with that; will do.

**Comment by jeffhandley** [comment]
The failing test is #51588

**Pull request merged**